Library/Homebrew/brew.rb

```diff
@@ -121,6 +121,9 @@ class MissingEnvironmentVariables < RuntimeError; end
     possible_tap = Tap.fetch(possible_tap.first) if possible_tap
 
     if !possible_tap || possible_tap.installed? || Tap.untapped_official_taps.include?(possible_tap.name)
+      if cmd == "cask" # Check for cask explicitly because it's very common in old guides
+        odie "`brew cask <command>` is no longer supported. Use `brew <command> --cask` instead."
+      end
```

Comment:
```suggestion
      # Check for cask explicitly because it's very common in old guides
      odie "`brew cask <command>` is no longer supported. Use `brew <command> --cask` instead." if cmd == "cask"
```

Reply:
```suggestion
      # Check for cask explicitly because it's very common in old guides
      odie "`brew cask` is no longer `brew` command. Use `brew <command> --cask` instead." if cmd == "cask"
```